### PR TITLE
Reset Ready condition when update failed

### PR DESCRIPTION
**Pull request by kajinamit**
This ensures the Ready Condition is reset when CR becomes once ready but something fails during reconfiguration.

This also fixes a few conditions unused or not initialized.

### CODE REVIEW

File: controllers/manila_controller.go [hunk 1]
```diff
@@ -138,11 +138,18 @@ func (r *ManilaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
...
+		if instance.Status.Conditions.AllSubConditionIsTrue() {
+			instance.Status.Conditions.MarkTrue(
+				condition.ReadyCondition, condition.ReadyMessage)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
```
Comment (abays) [inline controllers/manila_controller.go:151]: I think `manilaapi_controller.go` needs this sort of update as well.  Looks like the `ManilaAPI` in the CI deployment never reaches the `Ready` state without it, even though all the sub-conditions are themselves `Ready`: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/73/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1663842943112318976/artifacts/manila-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/api-resources/manilaapis.yaml

Comment (kajinamit) [reply controllers/manila_controller.go:151]: It's my bad that I thought I updated this but actually not. You are correct and this PR was incomplete and caused circular dependencies (IsReady is pending on ReadyCondition which is pending on IsReady)
I've updated the PR to cover ManilaAPI controller. Thanks for pointing that so quickly !

**Comment by abays** [comment]
> @kajinamit: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/manila-operator-build-deploy-tempest 	[24a9595](https://github.com/openstack-k8s-operators/manila-operator/commit/24a959573c4655704fc6f8de5d8ad0c713e58db7) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/73/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1663865952531386368) 	false 	`/test manila-operator-build-deploy-tempest`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=manila-operator&pr=73). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:kajinamit).

I think we need to remove this condition initialization line from `controllers/manila_controller.go`:

```
condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.DeploymentReadyInitMessage),
```

It's not used in that controller but prevents the new Ready logic from working when everything is otherwise actually ready.

**Comment by kajinamit** [comment]
@abays Thanks again. I've removed that unused DeploymentReadyCondition . I also noticed NetworkAttachemntsReadyCondition is not properly initialized in manila controller which I've fixed in the latest RP.

These are problems I've often found while distributing this change to all operators. Hope we fully fix this problem at this chance...

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**